engine-spring/core/src/main/java/org/camunda/bpm/engine/spring/SpringBeansResolverFactory.java

```diff
@@ -65,7 +71,19 @@ public boolean containsKey(Object key) {
   @Override
   public Object get(Object key) {
     if (key instanceof String) {
-      return applicationContext.getBean((String) key);
+      try {
+        return applicationContext.getBean((String) key);
+      } catch (BeanCreationException ex) {
+        if (SCOPE_NOT_ACTIVE.equals(ex.getClass().getName())) {
```

Comment:
When we had a similar change in another resolver, I remember, we added a little comment, like yours here. Maybe you could add a short comment in the code as well to explain why it's like that and when could we remove.

I think adding the `_EXCEPTION` to the constant would improve readability a little bit:

```suggestion
        if (SCOPE_NOT_ACTIVE_EXCEPTION.equals(ex.getClass().getName())) {
```